test/pylib/scylla_server.py

```diff
@@ -0,0 +1,307 @@
+
+    def __init__(self, exe=None, vardir=None, host_registry=None,
+                 cluster_name=None, seed=None, cmdline_options=None) -> None:
+        self.exe = pathlib.Path(exe).resolve()
+        self.vardir = pathlib.Path(vardir)
+        self.host_registry = host_registry
+        self.cmdline_options = cmdline_options
+        self.cfg = {
+            "cluster_name": cluster_name,
+            "host": None,
+            "seeds": seed,
+            "workdir": None,
+        }
+        self.cmd: Optional[Any] = None
+
+        async def stop_server():
+            if self.is_running:
+                await self.stop()
+
+        async def uninstall_server():
+            await self.uninstall()
+
+        self.stop_artifact = stop_server
+        self.uninstall_artifact = uninstall_server
+
+    async def install_and_start(self) -> None:
+        await self.install()
+
+        logging.info("starting server at host %s...", self.cfg["host"])
+
+        await self.start()
+
+        if self.cmd:
+            logging.info("started server at host %s, pid %d", self.cfg["host"], self.cmd.pid)
+
+    @property
+    def is_running(self) -> bool:
+        return self.cmd is not None
+
+    @property
+    def host(self) -> str:
+        return self.cfg["host"]
+
+    def find_scylla_executable(self) -> None:
+        if not os.access(self.exe, os.X_OK):
+            raise RuntimeError("{} is not executable", self.exe)
+
+    async def install(self) -> None:
+        """Create a working directory with all subdirectories, initialize
+        a configuration file."""
+
+        self.find_scylla_executable()
+
+        # Scylla assumes all instances of a cluster use the same port,
+        # so each instance needs an own IP address.
+        self.cfg["host"] = await self.host_registry.lease_host()
+        self.cfg["seeds"] = self.cfg.get("seeds") or self.cfg["host"]
+        # Use the last part in host IP 127.151.3.27 -> 27
+        # There can be no duplicates within the same test run
+        # thanks to how host registry registers subnets, and
+        # different runs use different vardirs.
+        self.shortname = pathlib.Path("scylla-" + self.cfg["host"].split(".")[-1])
+        self.cfg["workdir"] = self.vardir / self.shortname
+
+        logging.info("installing Scylla server in %s...", self.cfg["workdir"])
+
+        self.log_file_name = self.vardir / self.shortname.with_suffix(".log")
+
+        self.config_file_name = self.cfg["workdir"] / "conf/scylla.yaml"
+
+        # Use tmpdir (likely tmpfs) to speed up scylla start up, but create a
+        # link to it to keep everything in one place under testlog/
+        self.tmpdir = pathlib.Path(os.getenv('TMPDIR', '/tmp'))
+        self.tmpdir = self.tmpdir / ('scylla-'+self.cfg["host"])
+
+        # Cleanup any remains of the previously running server in this path
+        shutil.rmtree(self.tmpdir, ignore_errors=True)
+        pathlib.Path(self.cfg["workdir"]).unlink(missing_ok=True)
```

Comment:
You don't need to construct `Path` again, `self.cfg['workdir']` is already a `Path`.
Same for the 3 lines below.